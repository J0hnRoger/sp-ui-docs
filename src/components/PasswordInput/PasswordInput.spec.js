import React from 'react'
import renderer from 'react-test-renderer'
import PasswordInput from './PasswordInput'
import {shallow} from 'enzyme'

test('hides password quality by default', () => {
    // Create the three and serialize the result in JSON so we can compare it
    const tree = renderer.create(<PasswordInput
        htmlId="test"
        name="test"
        onChange={() => {}}
        value="Aubdz15#aA" />).toJSON()
    expect(tree).toMatchSnapshot() 
})

test('toggles input type when show/hide password clicked', () => {
    const wrapper = shallow(<PasswordInput
        htmlId="test"
        name="test"
        value="Aubdz15#aA" 
        onChange={() => {}}
        showVisibilityToggle />)
    
    expect(wrapper.find({type: 'password'})).toHaveLength(1)
    expect(wrapper.find({type: 'text'})).toHaveLength(0)
    
    wrapper.find('a').simulate('click')

    expect(wrapper.find({type: 'password'})).toHaveLength(0)
    expect(wrapper.find({type: 'text'})).toHaveLength(1)
  
})