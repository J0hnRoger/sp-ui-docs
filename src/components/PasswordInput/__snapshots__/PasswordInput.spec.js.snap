// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides password quality by default 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label>
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    placeholder={undefined}
    style={undefined}
    type="password"
    value="Aubdz15#aA"
  />
</div>
`;
